x/gamm/keeper/migrate.go

```diff
@@ -140,6 +165,49 @@ func (k Keeper) OverwriteMigrationRecords(ctx sdk.Context, migrationInfo types.M
 	}
 }
 
+// redirectDistributionRecord redirects the distribution record for the given balancer pool to the given concentrated pool.
+func (k Keeper) redirectDistributionRecord(ctx sdk.Context, cfmmPoolId, clPoolId uint64) error {
+	// Get CFMM gauges
+	cfmmGauges, err := k.poolIncentivesKeeper.GetGaugesForCFMMPool(ctx, cfmmPoolId)
+	if err != nil {
+		return err
+	}
+
+	if len(cfmmGauges) == 0 {
+		return fmt.Errorf("no gauges found for cfmm pool %d", cfmmPoolId)
+	}
+
+	// Get longest gauge duration from CFMM pool.
+	longestDurationGauge := cfmmGauges[0]
+	for i := 1; i < len(cfmmGauges); i++ {
+		if cfmmGauges[i].DistributeTo.Duration > longestDurationGauge.DistributeTo.Duration {
+			longestDurationGauge = cfmmGauges[i]
+		}
+	}
+
+	// Get concentrated liquidity gauge duration.
+	distributionEpochDuration := k.incentivesKeeper.GetEpochInfo(ctx).Duration
```

Comment:
Can you confirm that this is always meant to be equal epoch duration?

Reply:
Good call-out. As far as I can tell, this logic is solid to fall back on. See the `Distribute` function in the incentives module:

https://github.com/osmosis-labs/osmosis/blob/6a449dce9f2a6ffd8768d3394c47a7d4f9a3bbbd/x/incentives/keeper/distribute.go#L380-L393

This always distributes to the duration returned from the `GetEpochInfo` EpochInfo.